app/src/main/java/io/github/droidkaigi/confsched2017/viewmodel/SessionsViewModel.java

```diff
@@ -56,6 +60,8 @@ public void destroy() {
                     this.rooms = extractRooms(sessions);
                     this.stimes = extractStimes(sessions);
 
+                    notifyPropertyChanged(BR.loadingVisibility);
```

Comment:
Yep, I agree to update the visibility via DataBinding and to use progress bar!
My suggestion is to use `doOnSubscribe()`.

```xml
return Single.zip(sessionsRepository.findAll(languageId),
                mySessionsRepository.findAll(),
                (sessions, mySessions) -> {
                    final Map<Integer, MySession> mySessionMap = new LinkedHashMap<>();
                    for (MySession mySession : mySessions) {
                        mySessionMap.put(mySession.session.id, mySession);
                    }

                    this.rooms = extractRooms(sessions);
                    this.stimes = extractStimes(sessions);

                    List<SessionViewModel> viewModels = Stream.of(sessions)
                            .map(session -> {
                                boolean isMySession = mySessionMap.containsKey(session.id);
                                return new SessionViewModel(session, context, rooms.size(), isMySession);
                            })
                            .collect(Collectors.toList());
                    return adjustViewModels(viewModels, context);
                })
                .doOnSuccess(sessionViewModels -> {
                    setLoadingVisibile(false);
                })
                .doOnError(throwable -> {
                    setLoadingVisibile(false);
                })
                .doOnSubscribe(disposable -> {
                    setLoadingVisibile(true);
                })
```